src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -1818,6 +1818,13 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
     if (simdBaseJitType != CORINFO_TYPE_UNDEF)
     {
         simdBaseType = JitType2PreciseVarType(simdBaseJitType);
+#ifdef TARGET_XARCH
+        if (HWIntrinsicInfo::NeedsNormalizeSmallTypeToInt(intrinsic) && varTypeIsSmall(simdBaseType))
+        {
+            simdBaseJitType = varTypeIsUnsigned(simdBaseType) ? CORINFO_TYPE_UINT : CORINFO_TYPE_INT;
+            simdBaseType    = JitType2PreciseVarType(simdBaseJitType);
+        }
```

Comment:
This will catch normal importation but may miss nodes created elsewhere. In particular, we may correspondingly need to ensure that the type is normalized in `gtNewSimdBinOpNode`, `gtNewSimdTernaryLogicNode`, and any of the places in `lowerxarch` that are manually creating such nodes.

I expect we'd want an assert in `genHWIntrinsic` (`hwintrinsiccodegenxarch.cpp`) that similarly validates something like: `assert(!HWIntrinsicInfo::NeedsNormalizeSmallTypeToInt(intrinsic) || varTypeIsSmall(GetSimdBaseType()))` to help ensure that we didn't produce any such incorrectly typed nodes in earlier phases.

Reply:
`SSE2/Avx2/Avx512F.And/AndN/Or/Xor` should be exclusively created via `gtNewSimdBinOpNode` so I expect that's the only place to handle those.

`TernaryLogic` is created via `gtNewSimdTernaryLogicNode` in HIR and in a few places manually in Lowering.

`InsertVector256` and `ExtractVector256` are created in a few places in lowerxarch (such as from `GetLower`, `GetUpper`, `WithLower`, and `WithUpper`)